ecl/hql/hqlexpr.cpp

```diff
@@ -14681,8 +14681,9 @@ void TransformTrackingInfo::lock()
     if (curTransformDepth>=maxNestedLocks)
         maxNestedLocks = curTransformDepth+1;
 #endif
+    if (unlikely(((curTransformDepth+1) & TRANSFORM_DEPTH_MASK) == 0))
+        throw makeStringExceptionV(0, "INTERNAL ERROR: Transformers nested more than %u deep", TRANSFORM_DEPTH_MASK);
```

Comment:
What catches this exception?  Does the catcher log this error as an internal error?  Any reason not to have an error number for this?

Reply:
There will be an active catch handler which will report the exception as an error to the user.  I'm not concerned about an error number for an error that should never occur.